docs/introduction/reproducibility.md

```diff
@@ -0,0 +1,34 @@
+# Reproducibility
+
+A key point to service definition is the ability to reliably reproduce the exact same output over time. In case like
+
+1. A dependency or dependency of dependency was updated with an unintended breaking change
+2. Changes to a new version of a service shouldn't affect the older versions
+3. A change to the TypeSpec spec
+
+This can be mitigated with a few steps:
+
+## 1. Defend against dependencies changes
+
+_Note: This section applies if using `cadl install` or `npm install` to install dependencies. However other package manager (`yarn`, `pnpm`, etc.) have their own similar lock mechanism._
+
+When using `cadl install` or `npm install` a `package-lock.json` will be installed. This file SHOULD be committed to source control. It will ensure that later call to `cadl install` or `npm install` will use the exact versions resolved in the lock files unless the `package.json` was updated or a command like `npm update` was run.
```

Comment:
```suggestion
When using `cadl install` or `npm install` a `package-lock.json` will be installed. This file SHOULD be committed to source control. It will ensure that later calls to `cadl install` or `npm install` will use the exact versions resolved in the lock files unless the `package.json` was updated or a command like `npm update` was run.
```